src/lib/hooks/useGroups.ts

```diff
@@ -0,0 +1,26 @@
+import { useContext } from 'react';
+import { FirebaseContext } from '../../components/Firebase';
+import { QueryResult } from '../types';
+import useFirestoreQuery from './useFirestoreQuery';
+
+type GroupsDocumentData = {
```

Comment:
Do we prefer `type` over `interface`? (No need for change)

Reply:
I think I like to use types when it's something that's concrete and not meant to be shared or extended. Where as interfaces are more for shared types. In the end they are very similar in the latest TS versions, so I'm open to w/e

Reply:
I generally prefer `interface` but I'm not strongly for one or another unless there is a specific use-case